### DISCUSSION TITLE
BOSSA flashing on Arduino Nano 33 BLE (NRF52840)

### DISCUSSION THREAD

**Discussion by dlktdr** [zephyrproject-rtos/zephyr#36199] [Q&A]
I've been working on a project and am trying to use the Arduino variant of BOSSA to automatically reset it into bootloader mode with a 1200 baud touch.

I've tried including
```
CONFIG_BOOTLOADER_BOSSA_ARDUINO=y
CONFIG_BOOTLOADER_BOSSA_DEVICE_NAME="CDC_ACM_0"
```
In the prj.conf file. But I can't seem to get it to work this way.

Also tried to manually do it with the code copied from bossa.c in \soc\arm\atmel_sam0\common but it complains of missing reference cdc_acm_dte_rate_callback_set. Does the NRF52840 usb cdc driver not support the callback or am I doing something wrong here

**Comment by stephanosio** [comment]
BOSSA is for Atmel (now Microchip) SAM family devices. nRF52840 is not a SAM family device.

**Reply by dlktdr** [reply]
Arduino did a modified version of Bossa for their Nano 33 BLE from what I understand.

So I take it there is no support for the cdc_acm_dte_rate_callback_set on the NRF device?

**Reply by stephanosio** [reply]
> Arduino did a modified version of Bossa for their Nano 33 BLE from what I understand.

That's interesting. At least, it seems like that portion of code is not open source (couldn't find it anywhere online).

As for `cdc_acm_dte_rate_callback_set`, try setting `CONFIG_USB_CDC_ACM=y` and `CONFIG_CDC_ACM_DTE_RATE_CALLBACK_SUPPORT=y`.

**Reply by parasquid** [reply]
Sorry for the resurrect, but since this came up in my google search, just wanted to link the source of Arduino's BOSSA modification (not sure if this is everything and fully open source but the license does mention GPLv3)

https://github.com/arduino/BOSSA
https://github.com/shumatech/BOSSA/compare/master...arduino:BOSSA:nrf
